.github/workflows/amazon_bedrock.yml

```diff
@@ -68,15 +67,15 @@ jobs:
 
       # Do not authenticate on pull requests from forks
       - name: AWS authentication
+        id: aws-auth
         if: github.event.pull_request.head.repo.full_name == github.repository
         uses: aws-actions/configure-aws-credentials@b47578312673ae6fa5b5096b330d9fbac3d116df
         with:
           aws-region: ${{ env.AWS_REGION }}
           role-to-assume: ${{ secrets.AWS_CI_ROLE_ARN }}
 
-      # Do not run integration tests on pull requests from forks
       - name: Run integration tests
-        if: github.event.pull_request.head.repo.full_name == github.repository
+        if: success() && steps.aws-auth.outcome == 'success'
```

Comment:
This is just cosmetic.
I think it makes it clearer that we run integration tests only if authentication is executed and succeeds.